src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp

```diff
@@ -239,6 +239,34 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
     }
 };
 
+struct ArdentDefender : public AuraScript
+{
+    void OnAbsorb(Aura* aura, int32& currentAbsorb, int32& remainingDamage, uint32& /*reflectedSpellId*/, int32& /*reflectDamage*/, bool& preventedDeath) const override
+    {
+        currentAbsorb = 0;
+        Player* player = dynamic_cast<Player*>(aura->GetCaster());
+        if (!player)
+            return;
+        if (int32(player->GetHealth()) - remainingDamage > (player->GetMaxHealth() * 0.35))
+            return;
+        int32 reduction = 0;
+        int32 healMod = 0;
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_0)))
```

Comment:
Dont need to cast it to SpellEffectIndex anymore :P its redundant

Reply:
done + also added the defense factor (based on flimsy evidence)